src/main/kotlin/com/autonomousapps/tasks/AdviceTransformTask.kt

```diff
@@ -0,0 +1,98 @@
+@file:Suppress("UnstableApiUsage")
+
+package com.autonomousapps.tasks
+
+import com.autonomousapps.TASK_GROUP_DEP
+import com.autonomousapps.internal.advice.*
+import com.autonomousapps.internal.utils.*
+import com.autonomousapps.internal.utils.chatter
+import com.autonomousapps.tasks.AdviceTask.ConsoleReport
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.*
+import java.lang.StringBuilder
+
+/**
+ * Produces human- and machine-readable advice on how to modify a project's dependencies in order to have a healthy
+ * build.
+ */
+@CacheableTask
+abstract class AdviceTransformTask : DefaultTask() {
+
+  init {
+    group = TASK_GROUP_DEP
+    description = "Displays advice on screen"
+  }
+
+  @get:PathSensitive(PathSensitivity.RELATIVE)
```

Comment:
I know I use this in many places, but I think I'm wrong. I strongly suspect we can use `PathSensitivity.NONE`. So, this isn't a blocker, but if you get a chance to test NONE vs RELATIVE, that would be interesting.

Longer discussion: the documentation on this is very poor, but after a discussion with someone on the gradle community slack, I believe I understand it better. What `RELATIVE` means is that, if the path to the file input changes, then the input is considered changed -- even if the contents are the same. Since we don't _actually_ care about the path but only the content, I think `NONE` is appropriate. I have an open issue to go over all task inputs to check for this kind of thing, but I haven't had time yet. So this is more of an FYI / information-sharing comment.

Reply:
ok

Reply:
The doc is definitely poor of on this side.